store/src/directory.rs

```diff
@@ -392,21 +812,80 @@ where
                 Ok(v) if v.is_file() => {}
                 Ok(_) => continue,
             }
-            let ttl = match xattr::get(&path, format_xattr(crate::MetadataKey::<MKT>::Ttl.to_key()))
-            {
-                Err(e) => {
-                    log::trace!("Error looking up TTL xattr for {}: {:?}", path.display(), e);
+            // skip non-metadata files if xattr is not supported
+            if !self.xattr_enabled {
+                match path.as_path().extension() {
+                    None => continue,
+                    Some(ex) => {
+                        if ex != FDO_METADATA_EX {
+                            continue;
+                        }
+                    }
+                }
+                let path_meta = set_metadata_extension_to_path(&path)?;
+                let file_meta = match File::open(&path_meta) {
+                    Ok(f) => f,
+                    Err(e) if e.kind() == std::io::ErrorKind::NotFound => continue,
+                    Err(e) => {
+                        log::trace!(
+                            "Error opening metadata file {}: {:?}",
+                            path_meta.display(),
+                            e
+                        );
+                        continue;
+                    }
+                };
+                let metadata: FdoMetadata = match serde_cbor::from_reader(&file_meta) {
+                    Ok(data) => data,
+                    Err(e) => {
+                        return Err(StoreError::Unspecified(format!(
+                            "Error reading metadata from '{}': {:?}",
+                            path_meta.display(),
+                            e
+                        )))
+                    }
+                };
+                let val = metadata.map.get(crate::MetadataKey::<MKT>::Ttl.to_key());
+                match val {
+                    Some(ttl) => {
+                        if SystemTime::now() < ttl_from_disk(&ttl)? {
+                            continue;
+                        }
+                    }
+                    None => {
+                        continue;
+                    }
+                };
+                log::trace!("File at {} has expired, attempting removal", path.display());
+                if let Err(e) = fs::remove_file(&path) {
+                    log::warn!("Error deleting expired file {}: {}", path.display(), e);
+                }
+                if let Err(e) = fs::remove_file(&path_meta) {
```

Comment:
path and path_meta are the same here... as we're filtering out non-metadata files